src/pages/Home.jsx

```diff
@@ -1,26 +1,49 @@
+import { Link } from 'react-router-dom';
+import Alert from 'react-bootstrap/Alert';
 import { Wrapper } from '../components/styles/Wrapper.styled';
 import { MainTitle } from '../components/styles/MainTitle.styled';
 import { CardGrid } from '../components/styles/CardGrid.styled';
 import Card from '../components/Card';
-import { mockupVideosInfo } from '../mock-data/youtube-videos-mock';
+import Spinner from '../components/Spinner';
+import PropTypes from 'prop-types';
+
+const Home = ({ loading, error, videoData }) => {
+  if (error) {
+    console.error(error);
+    return <Alert variant="danger">An error occured.</Alert>;
+  }
 
-const Home = () => {
-  const videos = mockupVideosInfo.items;
   return (
     <Wrapper>
       <MainTitle>Home Page</MainTitle>
-      <CardGrid>
-        {videos.map((video) => (
-          <Card
-            key={video.etag}
-            thumbnail={video.snippet.thumbnails.medium.url}
-            title={video.snippet.title}
-            description={video.snippet.description}
-          />
-        ))}
-      </CardGrid>
+      {loading ? (
+        <Spinner />
+      ) : (
+        <CardGrid>
+          {videoData
+            ? videoData.items.map((video) => (
+                <Link
+                  to={`/video-details/${video.id.videoId}`}
+                  key={`${video.id.videoId}-${Math.random()}`}
+                >
```

Comment:
it will be great if this link is directly inside `Card` as it really is part of it